ipa-core/src/net/client/mod.rs

```diff
@@ -193,15 +193,17 @@ impl MpcHelperClient {
             };
             (
                 HttpsConnectorBuilder::new()
-                    .with_native_roots()
+                    .with_provider_and_native_roots(CRYPTO_PROVIDER.as_ref().clone())
```

Comment:
that's unfortunate - seems like an inconsistent API from rustls where some methods take `CryptoProvider` and some `Arc<CryptoProvider>` . Former immediately convert them to `Arc`, so really no reason why it shouldn't be `Arc<CryptoProvider>`